libraries/standard/mqtt/include/types/iot_mqtt_types.h

```diff
@@ -717,197 +835,146 @@ typedef struct IotMqttConnectInfo
-             * @param[in] size_t Number of elements in the subscription array.
-             * @param[out] uint8_t** Where the UNSUBSCRIBE packet is written.
-             * @param[out] size_t* Size of the UNSUBSCRIBE packet.
-             * @param[out] uint16_t* The packet identifier generated for this UNSUBSCRIBE.
              *
              * <b>Default implementation:</b> #_IotMqtt_SerializeUnsubscribe
              */
-            IotMqttError_t ( * unsubscribe )( const IotMqttSubscription_t * /* pSubscriptionList */,
-                                              size_t /* subscriptionCount */,
-                                              uint8_t ** /* pUnsubscribePacket */,
-                                              size_t * /* pPacketSize */,
-                                              uint16_t * /* pPacketIdentifier */ );
-
+            IotMqttSubscribeSerializer_t  unsubscribe;
             /**
              * @brief PINGREQ packet serializer function.
-             * @param[out] uint8_t** Where the PINGREQ packet is written.
-             * @param[out] size_t* Size of the PINGREQ packet.
              *
              * <b>Default implementation:</b> #_IotMqtt_SerializePingreq
              */
-            IotMqttError_t ( * pingreq )( uint8_t ** /* pPingreqPacket */,
-                                          size_t * /* pPacketSize */ );
-
+            IotMqttPingReqSerializer_t    pingreq;
             /**
              * @brief DISCONNECT packet serializer function.
-             * @param[out] uint8_t** Where the DISCONNECT packet is written.
-             * @param[out] size_t* Size of the DISCONNECT packet.
              *
              * <b>Default implementation:</b> #_IotMqtt_SerializeDisconnect
              */
-            IotMqttError_t ( * disconnect )( uint8_t ** /* pDisconnectPacket */,
-                                             size_t * /* pPacketSize */ );
+            IotMqttDisconnectSerializer_t disconnect;
         } serialize; /**< @brief Overrides the packet serialization functions for a single connection. */
 
         struct
         {
             /**
-             * @brief CONNACK packet deserializer function.
-             * @param[in,out] _mqttPacket* Pointer to an MQTT packet struct representing a CONNACK.
-             *
+             * @brief MQTT CONNACK packet deserializer function.
+             * @param[in,out] _mqttPacket* Pointer to an MQTT packet structure
+			 *
              * <b>Default implementation:</b> #_IotMqtt_DeserializeConnack
              */
-            IotMqttError_t ( * connack )( struct _mqttPacket * /* pConnack */ );
-
+            IotMqttDeserializer_t connack;
             /**
-             * @brief PUBLISH packet deserializer function.
-             * @param[in,out] _mqttPacket* Pointer to an MQTT packet struct representing a PUBLISH.
-             *
+             * @brief MQTT PUBLISH packet deserializer function.
```

Comment:
If you're adding the "MQTT" in front of the packet type, please add it to the serialize comments too.

Reply:
Removed the `MQTT` from these, it's redundant.